### PR TITLE
fix(e2e) Port OIDC tests to multiple connections COMPASS-8105

### PR DESCRIPTION
**Pull request by lerouxb**
Starting mongodb server like this:

(using https://github.com/lerouxb/start-oidc)

```
~/mongo/start-oidc % docker run -e EXPERIMENTAL_DOCKER_DESKTOP_FORCE_QEMU=1 --network host --platform linux/amd64 --rm -v .:/start-oidc -w /start-oidc node:20 bash -c 'npm i && npx ts-node start-oidc.ts "http://localhost:38231"'
```

This requires the experimental host networking feature enabled for mac and windows which requires signin to docker.

Running with:

```
~/mongo/compass/packages/compass-e2e-tests % OIDC_BIND_IP_ALL=true OIDC_CONNECTION_STRING="mongodb://127.0.0.1:27017/" OIDC_MOCK_HOSTNAME=localhost OIDC_MOCK_PORT=38231 npm run test-noserver-nocompile -- --test-multiple-connections
```

Obviously with or without `-- --test-multiple-connections` and also tweak `test-noserver-nocompile` to be either `test-noserver` or `test`. You probably want a `.only` on the describe too.

What this does is it starts the OIDC mock server listening on all ips so that the database server in docker can get to it. It starts it with a known hostname and port so that we can get a consistent, known issuer url to use in the mongodb server config. (38231 was just the last port picked automatically while I was working on this.). It then knows to use that mongodb server rather than start its own.

### CODE REVIEW

File: packages/compass-e2e-tests/tests/oidc.test.ts [hunk 1]
```diff
@@ -278,6 +309,11 @@ describe('OIDC integration', function () {
   });
 
   it('can successfully re-authenticate', async function () {
+    // TODO: not working for multiple connections
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/oidc.test.ts:312]: This doesn't pop up a modal or anything for multiple connections, but it does for single connection.

File: packages/compass-e2e-tests/tests/oidc.test.ts [hunk 2]
```diff
@@ -310,6 +352,11 @@ describe('OIDC integration', function () {
   });
 
   it('can decline re-authentication if wanted', async function () {
+    // TODO: not working for multiple connections
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/oidc.test.ts:355]: Same here.

File: packages/compass-e2e-tests/helpers/commands/hide-index.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export async function hideIndex(
     await browser.screenshot(screenshotName);
   }
 
-  await browser.clickVisible(Selectors.ConfirmationModalConfirmButton());
+  await browser.clickVisible(Selectors.confirmationModalConfirmButton());
```
Comment (lerouxb) [inline packages/compass-e2e-tests/helpers/commands/hide-index.ts:25]: drive-by: I cleaned up the naming. It is TitleCase for constants and camelCase for functions.

File: packages/compass-e2e-tests/tests/oidc.test.ts [hunk 4]
```diff
@@ -318,21 +358,36 @@ describe('OIDC integration', function () {
...
     );
     await browser.clickVisible(Selectors.ConnectButton);
 
-    const modal = '[role=dialog]';
-    const cancelButton = 'button:nth-of-type(2)';
-    await browser.waitUntil(async () => {
-      const modalHeader = await browser.$(`${modal} h1`);
-      return (await modalHeader.getText()).includes('Authentication expired');
-    });
+    await browser.pause(10_000);
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/tests/oidc.test.ts:372]: Maybe a comment here that this should aligned with the token expiration or something

**Review by gribnoysup — APPROVED** [review]
